src/averageAges.js

```diff
@@ -55,7 +57,13 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(human => onlyWithSon ? human.mother && human.sex === 'm' : human.mother)
+  const mothers = people.filter(human => people.some(child => child.mother === human.name));
+  const childMother = children.filter(child => mothers.some(mother=> child.mother === mother.name));
+  const averageAge = childMother.reduce((sum, curr) =>
+    sum + (curr.born - people
+      .find(mother => curr.mother === mother.name).born), 0) / childMother.length;
```

Comment:
Поформатуй трохи код, щоб підвищити читабельність
```suggestion
  const children = people.filter(
    human => onlyWithSon 
     ? human.mother && human.sex === 'm' 
     : human.mother
  );
  
  const mothers = people.filter(human => people.some(child => child.mother === human.name));
  const childMother = children.filter(child => mothers.some(mother=> child.mother === mother.name));
  const averageAge = childMother.reduce((sum, curr) =>
    sum + (curr.born - people
      .find(mother => curr.mother === mother.name).born), 0) / childMother.length;
```